### ISSUE TITLE
Add a feature to use dynamic linkage

### ISSUE THREAD

**Issue by jcjones** [dcuddeback/libudev-sys#1]
Thank you for maintaining this binding!

It'd be a nice add-on to support dynamic linkage for situations where static isn't doable. In [Firefox there's an example doing this with libpulse.so](http://searchfox.org/mozilla-central/source/media/libcubeb/cubeb-pulse-rs/pulse-ffi/src/ffi_funcs.rs#160) which I'm pretty sure was generated using [the dynamic/static `dlopen` feature support in bindgen](https://docs.rs/bindgen/0.25.4/bindgen/struct.Builder.html#method.link). I'll probably run bindgen myself to make a dynamic version, but it'd be great to have in the lib.

If you're interested, I or my team could produce a PR for this, but it'd probably be big and ugly from the code generator.

**Comment by ttaubert** [comment]
I don't think that bindgen actually supports dynamically loading shared libraries. The pulse-ffi code says it was generated but maybe it was only partially? Maybe only the function signatures when importing it first? [The history of the file](https://github.com/djg/cubeb-pulse-rs/commits/dev/pulse-ffi/src/ffi_funcs.rs) doesn't make it seems like there's a generator there, which leads me to believe that the "dlopen" feature was added manually.

**Comment by anisse** [comment]
Nowadays, bindgen supports dynamic loading. Would you be open to have a PR for this ?
